webknossos/webknossos/dataset/dataset.py

```diff
@@ -777,6 +792,10 @@ def add_layer(
 
         self._ensure_writable()
 
+        assert _ALLOWED_LAYER_NAME_REGEX.match(
+            layer_name
+        ), f"The layer name '{layer_name}' is invalid. It must only contain letters, numbers, underscores, hyphens and dots."
+
```

Comment:
could this also check the does-not-start-with-dort condition?

Reply:
Yes, the _ALLOWED_LAYER_NAME_REGEX checks that at least one allowed character that is not a dot has to be at the beginning of the string.